### DISCUSSION TITLE
Implementing strimzi kafka with LDAP

### DISCUSSION THREAD

**Discussion by bob3921** [strimzi/strimzi-kafka-operator#8834] [General]
Need help for the implementation with LDAP
We have strimzi Kafka provisioned also we have org. LDAP, Active Directory

Need to Implement it with our  Active directory.

Did go through the documents provided, but still wanted to understand what will be the steps needs to be followed, 
Please share the steps, related links

one of the link found : https://github.com/strimzi/strimzi-kafka-oauth

**Comment by scholzj** [comment]
I'm afraid we do not have any support for LDAP. But I think you can federate between your LDAP server and some OAuth server such as Keycloak and use OAuth authentication.

**Reply by bob3921** [reply]
Thanks for the update @scholzj 
So you mean that we cannot connect directly with it, it needs to have some ADFS(Like Keycloak is one of the option which can be used for this)
Then I think in that case I  need to understand how we can use the roles which are currently in place in AD to this Keycloak.

**Reply by scholzj** [reply]
I'm afraid I do not really have any details for this. I know this is what others used. I know this has some examples: https://github.com/keunlee/amq-streams-broker-authorization-sample - but not sure how it matches your use-case and how much up to date it is (AMQ Streams is a Red Hat product based on Strimzi, so that should not be a problem. But it could be that something changed since it was written)

**Reply by bob3921** [reply]
Thankyou @scholzj for your time on this.
